### PR TITLE
[MongoDB CDC] Fix “pollAwaitTimeMillis” does not take effect #1994

### PR DESCRIPTION
**Pull request by liangyaohui97**

### CODE REVIEW

File: flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/reader/fetch/MongoDBStreamFetchTask.java [hunk 1]
```diff
@@ -117,6 +117,7 @@ public void execute(Context context) throws Exception {
                 SourceRecord changeRecord = null;
                 if (!next.isPresent()) {
                     long untilNext = nextUpdate - time.milliseconds();
+                    nextUpdate += sourceConfig.getPollAwaitTimeMillis();
```
Comment (Jiabao-Sun) [inline flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/reader/fetch/MongoDBStreamFetchTask.java:120]: I think we should add adjust next update time under 
```
if (heartbeatManager != null) {
                        changeRecord =
                                heartbeatManager
                                        .heartbeat()
                                        .map(this::normalizeHeartbeatRecord)
                                        .orElse(null);
}
nextUpdate = time.milliseconds() + sourceConfig.getPollAwaitTimeMillis();
```

**Review by Jiabao-Sun — COMMENTED** [review]
Thanks @liangyaohui97 for this contribution.
I left few comments.

File: flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/reader/fetch/MongoDBStreamFetchTask.java (hunk 1 above)
Comment (liangyaohui97) [reply flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/reader/fetch/MongoDBStreamFetchTask.java]: Thank you for your guidance. This is really better. Previous settings may result in failure to penalize updated heartbeat。

**Review by Jiabao-Sun — APPROVED** [review]
Thanks @liangyaohui97 for the great work!
It LGTM in general. I made some changes about updating next update time because the heartbeat may also cost some time, using `current time + time interval` may be more accurate.

**Review by Jiabao-Sun — APPROVED** [review]

**Comment by Jiabao-Sun** [comment]
The use case of CI failure has nothing to do with this PR.
Merged into mater.

**Pull request merged**